lib/Auth/Source/privacyidea.php

```diff
@@ -234,6 +236,8 @@ protected function login_chal_resp($username, $password, $transaction_id, $signa
         /* If we get this far, we have a valid login. */
         $attributes = array();
         $arr = array("username", "surname", "email", "givenname", "mobile", "phone", "realm", "resolver");
+        // Add all additional attributes defined in the array map to the search array
+        $arr = array_merge(array_diff(array_keys($this->attributemap), $arr), $arr);
```

Comment:
According to http://php.net/manual/en/function.array-merge.php dublicate keys in the array will be overwritten. So I guess we can skip the *array_diff* and  and just do

    $arr = array_merge(array_keys($this->attributemap), $arr);

To my understanding there will be no dublicate values?